### PR TITLE
Precommit-hooks

### PR DESCRIPTION
**Pull request by ayinloya**
Run lint before code commit with pre-commit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,12 @@ To run the tests, run the following command:
 poetry run pytest
 ```
 
+Install [pre-commit](https://pre-commit.com/) to format code before committing.
+
+```shell
+pip install pre-commit
```
Comment (vanshg) [inline README.md]: Also mention the command needed to install the hook?

```suggestion
pip install pre-commit
pre-commit install
```

Comment (mldangelo) [reply README.md]: `pre-commit` should be included as a development dependency in pyproject.toml.

Comment (ayinloya) [reply README.md]: Nice catch, thanks

Comment (ayinloya) [reply README.md]: > `pre-commit` should be included as a development dependency in pyproject.toml.

Done

**Review by vanshg — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.4.1  # Use the sha / tag you want to point at
+    hooks:
+    -   id: mypy
+        # args: [--install-types,  --non-interactive, --ignore-missing-imports]
+-   repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+      - id: isort
+        name: isort (python)
```
Comment (mldangelo) [inline .pre-commit-config.yaml]: Adds trailing newline here

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+-   repo: https://github.com/PyCQA/pydocstyle
+    rev: 6.3.0
+    hooks:
+      - id: pydocstyle
+        args: [ --verbose, --convention=google]
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.4.1  # Use the sha / tag you want to point at
+    hooks:
+    -   id: mypy
+        # args: [--install-types,  --non-interactive, --ignore-missing-imports]
```
Comment (mldangelo) [inline .pre-commit-config.yaml]: Remove the commented out block or enable it

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+repos:
+-   repo: https://github.com/python-poetry/poetry
+    rev: 1.6.1
```
Comment (mldangelo) [inline .pre-commit-config.yaml:11]: Do we have to set rev’s here or can we rely on the ones set in the pyproject.toml file?

**Review by mldangelo — COMMENTED** [review]
Please see my nits. Thanks!

File: .pre-commit-config.yaml (hunk 4 above)
Comment (ayinloya) [reply .pre-commit-config.yaml:11]: I haven't found a way to do this yet

**Review by mldangelo — APPROVED** [review]
LGTM, thank you!

**Pull request merged**